### PR TITLE
dotnet-publish-iis refactoring

### PR DESCRIPTION
**Pull request by moozzyk**
- moving command logic to a dedicated class
- adding logic for reading webroot value from a parameter, hosting.json file and folder structure
- reading application name from project.json instead of a parameter

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -3,5 +3,7 @@
   <packageSources>
     <add key="AspNetVNext" value="https://www.myget.org/F/aspnetcidev/api/v2" />
     <add key="NuGet" value="https://api.nuget.org/v3/index.json" />
+    <add key="dotnet-cli" value="https://www.myget.org/F/dotnet-cli/api/v3/index.json" />
```
Comment (Tratcher) [inline NuGet.config]: I'm pretty sure we don't want these feeds here. @pranavkm?

File: src/dotnet-publish-iis/Program.cs [hunk 2]
```diff
@@ -15,46 +12,27 @@ public static int Main(string[] args)
...
+                Name = "dotnet publish-iis",
                 FullName = "Asp.Net IIS Publisher",
                 Description = "IIS Publisher for the Asp.Net web applications",
             };
             app.HelpOption("-h|--help");
 
-            var publishFolderOption = app.Option("--publish-folder", "The path to the publish output folder", CommandOptionType.SingleValue);
-            var appNameOption = app.Option("--application-name", "The name of the application", CommandOptionType.SingleValue);
+            var publishFolderOption = app.Option("--publish-folder|-p", "The path to the publish output folder", CommandOptionType.SingleValue);
+            var webRootOption = app.Option("--webroot|-w", "The name of webroot folder", CommandOptionType.SingleValue);
```
Comment (Tratcher) [inline src/dotnet-publish-iis/Program.cs]: If not the default wwwroot.

File: NuGet.config (hunk 1 above)
Comment (pranavkm) [inline NuGet.config]: Yeah - it messes up the gating.

**Comment by moozzyk** [comment]
I need to double check this - yesterday I was not able to build even dev branch without adding this feed. Maybe it was just a temporary problem?

File: src/dotnet-publish-iis/Program.cs (hunk 2 above)
Comment (moozzyk) [inline src/dotnet-publish-iis/Program.cs]: Not necessarily - webroot can also be read from hosting.json

File: NuGet.config (hunk 1 above)
Comment (moozzyk) [inline NuGet.config]: @pranavkm - not sure what it means apart from "remove this line"

Comment (Tratcher) [inline NuGet.config]: Remove both of these lines. The CI pulls these packages and pushes them to our own feeds.

**Comment by Tratcher** [comment]
The CoreClr and CLI builds have been broken on and off for weeks. That's why we have the CI pull them in and validate them.

File: NuGet.config (hunk 1 above)
Comment (pranavkm) [inline NuGet.config]: :+1:

Comment (moozzyk) [inline NuGet.config]: I may not be able to - I added `dotnet-core` to be able to get `"Microsoft.DotNet.ProjectModel": "1.0.0-*"`

**Comment by moozzyk** [comment]
Looks like today things work without additional feeds so I removed them from the NuGet.config file.

File: NuGet.config (hunk 1 above)
Comment (moozzyk) [inline NuGet.config]: Removed both entries

**Comment by Tratcher** [comment]
:shipit:

File: src/dotnet-publish-iis/Program.cs (hunk 2 above)
Comment (moozzyk) [inline src/dotnet-publish-iis/Program.cs]: @Tratcher : how about: 

The name of webroot folder. Overwrites webroot from hosting.json file or the default "wwwroot"

?

Comment (Tratcher) [inline src/dotnet-publish-iis/Program.cs]: To wordy for the command line. You can leave it as is.

Comment (moozzyk) [inline src/dotnet-publish-iis/Program.cs]: Yes. It's hard to something both explanatory and short. will leave as is then.

**Pull request merged**